remoulade/state/backend.py

```diff
@@ -0,0 +1,76 @@
+from collections import namedtuple
+from enum import Enum
+
+from ..encoder import Encoder
+from .errors import InvalidStateError
+
+
+class CatalogStates(Enum):
+    Pending = "Pending"
+    Success = "Success"
+    Skipped = "Skipped"
+    Started = "Started"
+    Canceled = "Canceled"
+    Failure = "Failure"
+
+    @classmethod
+    def has_value(cls, value):
+        return value in cls._value2member_map_
+
+
+#: A type alias representing states in the database
+class State(namedtuple("State", ("name", "args"))):
```

Comment:
I made some changes to State to make sure we serialize it and deserialize it well. 
```
class State(namedtuple("State", ("name", "args"))):
    """Catalog Class, that storages the state
        Parameters:
            name: Name of the state
            args: List of arguments in the state(name)
    """
    def __new__(cls, name, args):
        if name not in CatalogStates:
            raise InvalidStateError("The {} State is not defined".format(name))
        return super().__new__(cls, name, args)

    @classmethod
    def from_dict(cls, name, *args, **kwargs):
        return cls(name=CatalogStates[name], *args, **kwargs)

    def asdict(self):
        return {
            **self._asdict(),
            'name': self.name.value
        }
```

This way we are you to have a Enum in the state, and we should't need all these .name everywhere.
I'm not sure it should be "name" maybe "state" would be better ? what do you think ?

Reply:
super cool